apps/server/src/utils/time.ts

```diff
@@ -106,13 +107,48 @@ export const forgivingStringToMillis = (value, fillLeft = true) => {
  * @returns {number} - time in milliseconds
 
  */
-export const parseExcelDate = (excelDate) => {
+export const parseExcelDate = (excelDate: string): number => {
   // attempt converting to date object
   const date = new Date(excelDate);
-  if (date instanceof Date && !isNaN(date)) {
+  if (date instanceof Date && !isNaN(date.getTime())) {
     return dateToMillis(date);
   } else if (isTimeString(excelDate)) {
     return forgivingStringToMillis(excelDate);
   }
   return 0;
 };
+
+/**
+ * another go at simpler string formatting (counters) -- Copied from client code
+ * @description Converts seconds to string representing time
+ * @param {number | null} milliseconds - time in seconds
+ * @param {boolean} [hideZero] - whether to show hours in case its 00
+ * @returns {string} String representing absolute time 00:12:02
+ */
+export function formatDisplay(milliseconds: number | null, hideZero = false): string {
+  if (typeof milliseconds !== 'number') {
+    return hideZero ? '00:00' : '00:00:00';
+  }
+
+  // add an extra 0 if necessary
+  const format = (val: number) => `0${Math.floor(val)}`.slice(-2);
+
+  const s = Math.abs(millisToSeconds(milliseconds));
+  const hours = Math.floor((s / 3600) % 24);
+  const minutes = Math.floor((s % 3600) / 60);
+
+  if (hideZero && hours < 1) return [minutes, s % 60].map(format).join(':');
+  return [hours, minutes, s % 60].map(format).join(':');
+}
+
+/**
+ * @description Converts milliseconds to seconds -- Copied from client code
+ * @param {number | null} millis - time in seconds
+ * @returns {number} Amount in seconds
+ */
+export const millisToSeconds = (millis: number | null): number => {
```

Comment:
are this copies of the ones in the client? should we move them to the shared code instead?
Also, maybe we should leverage luxon for the parsing instead?
I can see people start asking for AM PM stuff next